polynote-server/src/main/scala/polynote/server/SocketSession.scala

```diff
@@ -85,12 +87,35 @@ object SocketSession {
       serverVersion = BuildInfo.version,
       serverCommit = BuildInfo.commit,
       identity = identity.map(i => Identity(i.name, i.avatar.map(ShortString))),
-      sparkTemplates = config.spark.flatMap(_.propertySets).getOrElse(Nil)
+      sparkTemplates = config.spark.flatMap(_.propertySets).getOrElse(Nil),
+      notebookTemplates = config.behavior.notebookTemplates
     )
 
   def getRunningKernels: RIO[SessionEnv with PublishMessage with NotebookManager, RunningKernels] = for {
     paths          <- NotebookManager.listRunning()
     statuses       <- ZIO.collectAllPar(paths.map(NotebookManager.status))
     kernelStatuses  = paths.zip(statuses).map { case (p, s) => ShortString(p) -> s }
   } yield RunningKernels(kernelStatuses)
+
+  def getMaybeContent(maybeContent: Option[String], maybeTemplatePath: Option[String]): ZIO[Config, Throwable, Option[String]] = {
+    maybeContent match {
+      case Some(content) => ZIO.succeed(Some(content))
+      case None => for {
+        config  <- Config.access.map(_.behavior.notebookTemplates)
+        content <- maybeTemplatePath match {
+          case Some(templatePath) if config.contains(templatePath) => readFromTemplatePath(templatePath)
```

Comment:
Should probably throw an exception (and log) if a template path was specified but it's not in the config. Would likely be some kind of exploit attempt.

Reply:
E.g. `case Some(templatePath) => ZIO.fail(new IllegalArgumentException(s"$templatePath is not a configured notebook template"))`